### ISSUE TITLE
Check local directory exists before doing remote check

### ISSUE THREAD

**Issue by andyhasit** [ncw/rclone#1067]
I attempted to clone a directory on path on a mounted drive which normally works, but on this occasion it didn't  because the drive wasn't mounted yet, as I hadn't tried to access it since booting up. So I got this error:

Local file system at /media/andrew/A4AAD085AAD05600/WIN_DEV: directory not found

No problem there, but I think it would better if rclone did basic checks on the local directory FIRST, instead of after doing all the checks on the remote, which can take a very long time. I'm presuming the same sequence applies if there are other issues with the local directory such as permissions, so this enhancement would just save a bit of frustration, on some occasions.

Low priority of course.

> What is your rclone version (eg output from `rclone -V`)
rclone v1.35
> Which OS you are using and how many bits (eg Windows 7, 64 bit)
Linux mint 18.1 64bit
> Which cloud storage system are you using? (eg Google Drive)
Google Drive
> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
rclone sync /media/andrew/A4AAD085AAD05600/WIN_DEV GoogleDrive:backup/WIN_DEV --exclude-from /home/andrew/Documents/backup/rclone-ignore.txt
> A log from the command with the `-v` flag (eg output from `rclone -v copy /tmp remote:tmp`)
Local file system at /media/andrew/A4AAD085AAD05600/WIN_DEV: directory not found

**Comment by ncw** [comment]
Yes you are right...

However this will be fixed by #517 which is in progress :-)

**Comment by ncw** [comment]
I have merged this to master now - see here for a beta:

http://beta.rclone.org/v1.35-74-g48cdedc/

Any feedback much appreciated

**Comment by andyhasit** [comment]
Using rclone from the link you provided, this is the output I get: 

```sh
/home/andrew/Desktop/rclone sync /media/andrew/A4AAD085AAD05600/WIN_DEV GoogleDrive:backup/WIN_DEV --exclude-from /home/andrew/Documents/backup/rclone-ignore.txt
2017/02/16 12:02:07 Google drive root 'backup/WIN_DEV': Waiting for checks to finish
2017/02/16 12:02:07 Google drive root 'backup/WIN_DEV': Waiting for transfers to finish
2017/02/16 12:02:07 Google drive root 'backup/WIN_DEV': not deleting files as there were IO errors
2017/02/16 12:02:07 Attempt 1/3 failed with 0 errors and: error reading source directory "": directory not found
2017/02/16 12:02:08 Google drive root 'backup/WIN_DEV': Waiting for checks to finish
2017/02/16 12:02:08 Google drive root 'backup/WIN_DEV': Waiting for transfers to finish
2017/02/16 12:02:08 Google drive root 'backup/WIN_DEV': not deleting files as there were IO errors
2017/02/16 12:02:08 Attempt 2/3 failed with 0 errors and: error reading source directory "": directory not found
2017/02/16 12:02:08 Google drive root 'backup/WIN_DEV': Waiting for checks to finish
2017/02/16 12:02:08 Google drive root 'backup/WIN_DEV': Waiting for transfers to finish
2017/02/16 12:02:08 Google drive root 'backup/WIN_DEV': not deleting files as there were IO errors
2017/02/16 12:02:08 Attempt 3/3 failed with 0 errors and: error reading source directory "": directory not found
2017/02/16 12:02:08 Failed to sync: error reading source directory "": directory not found
```

So it does fail early, but it still seems to be printing out info about the remote process whereas I'd have thought it should just say something like:

```
Local directory not found:
/media/andrew/A4AAD085AAD05600/WIN_DEV
Aborting backup.
```

**Comment by ncw** [comment]
rclone could check that the source directory exists before starting the sync, but that would slow things down in the general case because it needs another transaction.  It will get an error as soon as it starts to use the directory which in the new code is quite quick.

I'm planning on rejiggling the internals of the remotes which will give me a chance to do this properly so I'll put it on the list!
